packages/webdriverio/src/commands/element/click.ts

```diff
@@ -123,6 +124,7 @@ export async function click(
 
     if (this.isW3C) {
         const browser = getBrowserObject(this)
+        await moveInViewport(this, 2000)
```

Comment:
Let's just scroll to it using scrollIntoView. Checking if the element is in the viewport is just as expensive of an operation so in the best case you only do the check but in the worst you do the check, then scroll. Having to do both for multiple elements is expensive. Especially when running in a cloud environment where the roundtrip takes a very long time

Reply:
I agree action like waiting for animation could be omitted
```
await browser.waitUntil(async ()=> await isInViewport(elm), { timeout: scrollTimeout })
```
but rest of them still usefull because why we should move it if it's in intoview ?

by default web version of scrollIntoView with the option like 'nearest' have the check, into view or not, under hood, but wdio implementation using action if it's not failed. That's why it's going to be called each time,

Reply:
Using {block: "start"}, the element is aligned at the top of its ancestor.
Using {block: "center"}, the element is aligned at the middle of its ancestor.
Using {block: "end"}, the element is aligned at the bottom of its ancestor.

Using {block: "nearest"}, the element:
is aligned at the top of its ancestor if you're currently below it.
is aligned at the bottom of its ancestor if you're currently above it.
stays put, if it's already in view.

Reply:
I know, I have refactored the scrollIntoView command not too long ago to verify all this behaviour.

If we are going to send a script to the server it doesn't matter if you send the check or the scroll itself. Scrolling or checking if the element is in the viewport both take about the same amount of time so you gain nothing by first doing the check in my opinion.
I understand why you introduced the check but I think it is better to leave it out and just try to scroll each element into the viewport as it's instant with the new scrollIntoView implementation anyway.

Reply:
Having that said, if Christian is fine with the way you implemented it. I am fine with resolving this, I just want to share what I feel would be the best approach 🙂

Reply:
I got it... ok maybe it's worth to delete first check but still have the second ?

Reply:
async function moveInViewport (elm: WebdriverIO.Element) {
        await  elm.scrollIntoView({ block : 'nearest', inline: 'nearest', behavior: 'instant' })

        if (await isInViewport(elm)) {
            await  elm.scrollIntoView({ block : 'center', inline: 'center', behavior: 'instant' })
        }
}

Reply:
Moreover I'd suggest use it only for moveTo

Reply:
there's already a retry built into the scrollIntoView so this is not needed.